### PR TITLE
Audio/background player for downloads

### PR DESCRIPTION
**Pull request by Bnyro**

### CODE REVIEW

File: app/src/main/java/com/github/libretube/services/OfflinePlayerService.kt [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+import com.google.android.exoplayer2.MediaItem
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+
+/**
+ * A service to play downloaded audio in the background
+ */
+class OfflinePlayerService : Service() {
```
Comment (Isira-Seneviratne) [inline app/src/main/java/com/github/libretube/services/OfflinePlayerService.kt:29]:
```suggestion
class OfflinePlayerService : LifecycleService() {
```

File: app/src/main/java/com/github/libretube/services/OfflinePlayerService.kt [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+            .setSmallIcon(R.drawable.ic_launcher_lockscreen)
+            .build()
+
+        startForeground(PLAYER_NOTIFICATION_ID, notification)
+    }
+
+    override fun onStartCommand(intent: Intent, flags: Int, startId: Int): Int {
+        val videoId = intent.getStringExtra(IntentData.videoId)
+
+        CoroutineScope(Dispatchers.IO).launch {
```
Comment (Isira-Seneviratne) [inline app/src/main/java/com/github/libretube/services/OfflinePlayerService.kt:48]:
```suggestion
        lifecycleScope.launch(Dispatchers.IO) {
```

**Review by Isira-Seneviratne — CHANGES_REQUESTED** [review]
Overall, this looks good to me. I've suggested a couple of small changes.

**Comment by mxwmnn** [comment]
Works partially, you can't skip songs it only plays one audio and doesn't have continuous play. Would be nice. Thanks for the effort guys

**Pull request merged**